dts/bindings/input/zephyr,input-lvgl.yaml

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 Fabian Blatz <fabianblatz@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  LVGL indev pseudo-device
+
+  Listens for input events and routes the
+  lv_indev_data_t to the underlying lv_indev_t managed by LVGL.
+
+  Needs to be associated to a specific device to listen for events
+  from that device. Example configuration:
+
+  pointer {
+          compatible = "zephyr,input-lvgl";
+          input = <&input_sdl_touch>;
+          type = "pointer";
+  };
+
+
+compatible: "zephyr,input-lvgl"
+
+properties:
+  input:
+    required: true
+    type: phandle
+    description: |
+      Input device phandle, if not specified listen for input from all devices.
+
+  type:
+    type: string
+    required: true
+    enum:
+      - "pointer"
+      - "keypad"
+      - "button"
+      - "encoder"
+    description: LVGL indev type.
+  
+  msgq-size:
```

Comment:
I'm a bit worried about the proliferation of "software devices" in devicetree, in particular here we are using devicetree to define software-specific things like a message queue size.

I think this needs discussion, and potentially alternative solutions. This was not needed by e.g. kscan, so why is this added for input?

Reply:
msgq-size should probably remain an option, but the various swap-... I think belong to here since they are a hardware characteristic, plus you could imagine supporting multiple devices with different setups.

The other thing that should go here is the mapping of axis and button codes, which matches the ones emitted by a `gpio-keys` for example, so I think it makes sense for those too.

Reply:
> devicetree to define software-specific things like a message queue size.

Agreed, I will move the `msgq-size` back to being a Kconfig option. 

> This was not needed by e.g. kscan, so why is this added for input?

Yeah, the way LVGL expects to be fed input event needs a message queue of some sorts. Previously this was a Kconfig option in the LVGL module.

> The other thing that should go here is the mapping of axis and button codes

For the axis I am not sure. Lots of LVGL code either makes use of the `enc_diff` value or it expects that the encoder really only emits `LV_KEY_LEFT` or `LV_KEY_RIGHT`: [https://docs.lvgl.io/master/overview/indev.html](https://docs.lvgl.io/master/overview/indev.html), e.g. [lv_arc](https://github.com/lvgl/lvgl/blob/v8.3.7/src/widgets/lv_arc.c#L554) . I suppose I'd need to set the `lv_indev_data_t->code` depending on the `input_event->value` sign what do you think?

Reply:
> For the axis I am not sure. Lots of LVGL code either makes use of the `enc_diff` value or it expects that the encoder really only emits `LV_KEY_LEFT` or `LV_KEY_RIGHT`: https://docs.lvgl.io/master/overview/indev.html, e.g. [lv_arc](https://github.com/lvgl/lvgl/blob/v8.3.7/src/widgets/lv_arc.c#L554) . I suppose I'd need to set the `lv_indev_data_t->code` depending on the `input_event->value` sign what do you think?

Right but you need to decide what event code to filter on, for example `INPUT_REL_Z`. For touchscreen all drivers send X/Y but for wheels and buttons it's up to the driver and you want to match it so we need to tell the LVGL input code somehow, and I think it makes sense to have it in the dts file.

Reply:
Ah ok, so this was meant for filtering purposes. I thought you meant adding the axis data somehow into `lv_indev_data_t`.

Reply:
Yeah, or any other configuration that depends on the hardware setup, I imagine stuff like `btn_points` data could go there.